core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -107,33 +115,6 @@
   @Override
   public void start(Listener<RespT> observer, Metadata headers) {
     Preconditions.checkState(stream == null, "Already started");
-    Long deadlineNanoTime = callOptions.getDeadlineNanoTime();
```

Comment:
I made an optimization per the feedback from @ejona86, that `ClientCallImpl.start()` will skip `DelayedStream` when possible. Both `ClientCallImpl.start()` and `DelayedStream` now could create the stream on the transport, where timeout needs to be set to headers. The altering of other headers must remain in `ClientCallImpl.start()`. So they cannot be in one place.